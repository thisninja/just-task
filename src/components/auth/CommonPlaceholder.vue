<script>
export default {
  functional: true,
  props: ['title'],
  render(h, { props }) {
    return [
      h('img', {
        class: 'layout__icon',
        attrs: {
          src: 'src/assets/task-icon.jpg',
        },
      }),
      h('h3', props.title)
    ];
  }
}
</script>
