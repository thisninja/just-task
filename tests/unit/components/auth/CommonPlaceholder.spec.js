import { mount } from '@vue/test-utils';
import CommonPlaceholder from '@/components/auth/CommonPlaceholder.vue';

describe('CommonPlaceholder.vue', () => {
  const title = 'bar';

  const WrapperHOC = {
    components: { CommonPlaceholder },
    template: `
    <div>
      <common-placeholder v-bind="$attrs" />
    </div>
  `
  }

  const wrapper = mount(WrapperHOC, {
    propsData: {
      title,
    },
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
