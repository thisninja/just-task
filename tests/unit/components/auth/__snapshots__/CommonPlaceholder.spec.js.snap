// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonPlaceholder.vue should match snapshot 1`] = `
<div title="bar"><img src="src/assets/task-icon.jpg" class="layout__icon">
  <h3>bar</h3>
</div>
`;
